### PR TITLE
[POC] Create layout helper for UIKit

### PR DESCRIPTION
**Pull request by jacklyn-situmorang**
This PR is adding layout helper for UIKit. An example of using `addSubviewSizedEqually` can be found on `CheckboxViewController` and an example of using `addConstraint` can be found on `Badge+UIPresentable`

### CODE REVIEW

File: core/Sources/Common/UIKit/Extension/UIView/UIView+Layout.swift [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+//  Copyright © 2023 Adevinta. All rights reserved.
+//
+
+import UIKit
+
+extension UIView {
+
+    /// Adds a subview with the same size as the view.
+    /// - Parameter subview: subview to be added
+    public func addSubviewSizedEqually(_ subview: UIView) {
```
Comment (LouisBorleeAdevinta) [inline core/Sources/Common/UIKit/Extension/UIView/UIView+Layout.swift:15]: I don't think it should be public ⚠️

File: spark/Demo/Classes/View/Components/Checkbox/UIKit/CheckboxViewController.swift [hunk 2]
```diff
@@ -76,17 +76,8 @@ final class CheckboxViewController: UIViewController {
     override func viewDidLoad() {
         super.viewDidLoad()
 
-        view.addSubview(self.scrollView)
-        self.scrollView.leadingAnchor.constraint(equalTo: view.leadingAnchor).isActive = true
-        self.scrollView.trailingAnchor.constraint(equalTo: view.trailingAnchor).isActive = true
-        self.scrollView.topAnchor.constraint(equalTo: view.topAnchor).isActive = true
-        self.scrollView.bottomAnchor.constraint(equalTo: view.bottomAnchor).isActive = true
-
-        self.scrollView.addSubview(self.contentView)
```
[the remark this answers is not in the document]
Comment (jacklyn-situmorang) [reply spark/Demo/Classes/View/Components/Checkbox/UIKit/CheckboxViewController.swift:85]: You're right, the `addSubviewSizedEqually` only works when adding the scrollView. I'll restore the old version.

File: core/Sources/Common/UIKit/Extension/UIView/UIView+Layout.swift [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        addSubview(subview)
+
+        NSLayoutConstraint.activate([
+            subview.leadingAnchor.constraint(equalTo: self.leadingAnchor),
+            subview.trailingAnchor.constraint(equalTo: self.trailingAnchor),
+            subview.topAnchor.constraint(equalTo: self.topAnchor),
+            subview.bottomAnchor.constraint(equalTo: self.bottomAnchor),
+            subview.centerXAnchor.constraint(equalTo: self.centerXAnchor),
```
[the remark this answers is not in the document]
Comment (robergro) [reply core/Sources/Common/UIKit/Extension/UIView/UIView+Layout.swift]: And you can use the NSLayoutConstraint.stickEdges to just do that on one line :) (NSLayoutConstraint extension file on SparkCore)

File: core/Sources/Common/UIKit/Extension/UIView/UIView+Layout.swift [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+        self.translatesAutoresizingMaskIntoConstraints = false
+        subview.translatesAutoresizingMaskIntoConstraints = false
+        addSubview(subview)
+
+        NSLayoutConstraint.activate([
+            subview.leadingAnchor.constraint(equalTo: self.leadingAnchor),
+            subview.trailingAnchor.constraint(equalTo: self.trailingAnchor),
+            subview.topAnchor.constraint(equalTo: self.topAnchor),
+            subview.bottomAnchor.constraint(equalTo: self.bottomAnchor)
+        ])
```
Comment (michael-zimmermann) [inline core/Sources/Common/UIKit/Extension/UIView/UIView+Layout.swift:25]:
```suggestion
        NSLayoutConstraint.stickEdges(from: self, to: subview)
```

**Review by michael-zimmermann — APPROVED** [review]
👍

**Pull request merged**